internal/services/appservice/helpers/windows_web_app_schema.go

```diff
@@ -0,0 +1,768 @@
+			}
+		}
+	}
+	expanded.AlwaysOn = pointer.To(winSiteConfig.AlwaysOn)
+
+	if metadata.ResourceData.HasChange("site_config.0.api_management_api_id") {
+		expanded.APIManagementConfig = &web.APIManagementConfig{
+			ID: pointer.To(winSiteConfig.ApiManagementConfigId),
+		}
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.api_definition_url") {
+		expanded.APIDefinition = &web.APIDefinitionInfo{
+			URL: pointer.To(winSiteConfig.ApiDefinition),
+		}
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.app_command_line") {
+		expanded.AppCommandLine = pointer.To(winSiteConfig.AppCommandLine)
+	}
+
+	if metadata.ResourceData.HasChange("site_config.0.application_stack") {
+		if len(winSiteConfig.ApplicationStack) == 1 {
+			winAppStack := winSiteConfig.ApplicationStack[0]
+			// TODO - only one of these should be non-nil?
+			if winAppStack.NetFrameworkVersion != "" {
+				expanded.NetFrameworkVersion = pointer.To(winAppStack.NetFrameworkVersion)
+				if currentStack == "" {
+					currentStack = CurrentStackDotNet
+				}
+			}
+			if winAppStack.NetCoreVersion != "" {
+				expanded.NetFrameworkVersion = pointer.To(winAppStack.NetCoreVersion)
+				if currentStack == "" {
+					currentStack = CurrentStackDotNetCore
+				}
+			}
+			if winAppStack.NodeVersion != "" {
+				// Note: node version is now exclusively controlled via app_setting.WEBSITE_NODE_DEFAULT_VERSION
+				if currentStack == "" {
+					currentStack = CurrentStackNode
+				}
+			}
+			if winAppStack.PhpVersion != "" {
+				if winAppStack.PhpVersion != PhpVersionOff {
+					expanded.PhpVersion = pointer.To(winAppStack.PhpVersion)
+				} else {
+					expanded.PhpVersion = pointer.To("")
+				}
+				if currentStack == "" {
+					currentStack = CurrentStackPhp
+				}
+			}
+			if winAppStack.PythonVersion != "" || winAppStack.Python {
+				expanded.PythonVersion = pointer.To(winAppStack.PythonVersion)
+				if currentStack == "" {
+					currentStack = CurrentStackPython
+				}
+			}
+			if winAppStack.JavaVersion != "" {
+				expanded.JavaVersion = pointer.To(winAppStack.JavaVersion)
+				switch {
+				case winAppStack.JavaEmbeddedServer:
+					expanded.JavaContainer = pointer.To(JavaContainerEmbeddedServer)
+					expanded.JavaContainerVersion = pointer.To(JavaContainerEmbeddedServerVersion)
+				case winAppStack.TomcatVersion != "":
+					expanded.JavaContainer = pointer.To(JavaContainerTomcat)
+					expanded.JavaContainerVersion = pointer.To(winAppStack.TomcatVersion)
+				case winAppStack.JavaContainer != "":
+					expanded.JavaContainer = pointer.To(winAppStack.JavaContainer)
+					expanded.JavaContainerVersion = pointer.To(winAppStack.JavaContainerVersion)
+				}
+
+				if currentStack == "" {
+					currentStack = CurrentStackJava
+				}
+			}
+			if winAppStack.DockerContainerName != "" || winAppStack.DockerContainerRegistry != "" || winAppStack.DockerContainerTag != "" {
+				if winAppStack.DockerContainerRegistry != "" {
```

Comment:
The logic for 
```
if winAppStack.DockerContainerName != "" || winAppStack.DockerContainerRegistry != "" || winAppStack.DockerContainerTag != "" {
	if winAppStack.DockerContainerRegistry != "" {
```
means that `winAppStack.DockerContainerName` and `winAppStack.DockerContainerTag` could be empty when trying to set them
```
expanded.WindowsFxVersion = pointer.To(fmt.Sprintf("DOCKER|%s/%s:%s", winAppStack.DockerContainerRegistry, winAppStack.DockerContainerName, winAppStack.DockerContainerTag))
```

Do we need to check that those aren't empty?

Reply:
They can be empty in some cases - this is an open issue and I'm planning on working on the Docker support in a follow-up PR due to the size of this one already, and the potential complexity of supported configurations for Docker (e.g. docker-compose, and supported image reference formats etc)